spring-boot-project/spring-boot-docs/src/docs/asciidoc/configuration-metadata/annotation-processor.adoc

```diff
@@ -33,7 +33,9 @@ If you are using an `additional-spring-configuration-metadata.json` file, the `c
 
 [source,gradle,indent=0,subs="verbatim"]
 ----
-	compileJava.inputs.files(processResources)
+	tasks.named('compileJava') {
+	    inputs.files(processResources)
```

Comment:
I think will eagerly create the `processResources` task.

Reply:
This will only create the `processResources` task when the closure is executed, which in turn only happens when `compileJava` is being created/configured. In this case, `processResources` would have to be created anyway.

Reply:
I wonder if that is that the case 100% of the time.

Regardless, this is an example of the increased complexity of task configuration avoidance — you need to be more aware of the relationships between tasks. According to Gradle’s own docs, you also need to [consider where the task was created](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:task_configuration_avoidance_pitfalls).

In this case `processResources` is defined by another plugin, but so is `compileJava`. That makes Gradle’s guidance a little unclear to me.

My feeling is that consistently using tasks.named in a situation such as this reduces cognitive load. You don’t need to think about where a task has come from or their existing relationships and things will just work (I think).

Reply:
In my tests it did not make a difference if `processResources` is used lazily or directly (inside the lazily evaluated `compileJava` closure). The nearest match in the Gradle docs I could find is this:

```gradle
tasks.register("someTask")

tasks.register("anEagerTask") {
    dependsOn someTask
}
```
Here `someTask` is referenced directly inside the lazy configuration closure of `anEagerTask`.

That being said, I agree that just always using `tasks.named` makes things more consistent.